### DISCUSSION TITLE
The Save Workspace button is not available(Gray) when i manually layout without click render

### DISCUSSION THREAD

**Discussion by cavies** [structurizr/onpremises#20] [Q&A] [answered]
Version 2956：The Save Workspace button is not available when i manually layout without click render

**Comment by simonbrowndotje** [comment]
I don't see this myself. If I load the DSL editor and move an element in the embedded diagram viewer, the save button immediately turns red.

**Reply by cavies** [reply]
I posted the information I saw on the JavaScript console, and it looks like it's caused by this error
![image](https://user-images.githubusercontent.com/15722487/219589403-81a665ed-d66c-4d13-be1e-6ed0ab6dbf05.png)

**Reply by cavies** [reply]
I cleaned the cache and re-pulled the corresponding javascript file, and the problem was solved. I upgraded from 2920 to 2956 today, is it because there are some differences in the JavaScript files in the two versions?

**Comment by simonbrowndotje** [comment, marked as answer]
> are some differences in the JavaScript files in the two versions?

Yes.
